### PR TITLE
custom SIGP model save & load

### LINKED ISSUE #68: Difficulty saving and loading models using`NonTensorialInputs` data
First off, great work, this is a really cool package!

I've been playing with the graph representation inputs using `graphein` to a model building off of `SIGP` (some examples in your codebase call it `GraphGP`) and have been getting some really great performance out of it. However, I'm struggling to understand how to correctly save and then load the model back into memory for inference after training. If I save the state dict then re-init using that state dict, the model performs as if it had been randomly initialized. I also tried pickling the model (not the ideal solution) I get the following exception:
```python
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[88], [line 4](vscode-notebook-cell:?execution_count=88&line=4)
      [1](vscode-notebook-cell:?execution_count=88&line=1) import pickle
      [3](vscode-notebook-cell:?execution_count=88&line=3) with open('model.pkl', 'wb') as file:
----> [4](vscode-notebook-cell:?execution_count=88&line=4)     pickle.dump(model, file)

RuntimeError: Pickling of "rdkit.Chem.rdchem.Atom" instances is not enabled (http://www.boost.org/libs/python/doc/v2/pickle.html)
```
I tried setting `train_inputs` to `None` before saving. This took care of the exception, however I'm back to the original issue where the model seems to be randomly initialized.

I was wondering if you had any guidance here, or if there was something in the docs that I missed. Thanks!

### PR DESCRIPTION
**Pull request by kkovary**
#### Reference Issues/PRs

Fixes #68

#### Pull Request Checklist

- [ ] Added a note about the modification or contribution to the `./CHANGELOG.md` file (if applicable)
- [ ] Added appropriate unit test functions in the `./gauche/tests/*` directories (if applicable)
- [ ] Modify documentation in the corresponding Jupyter Notebook under `./notebooks/` (if applicable)
- [ ] Ran `python -m py.test tests/` and make sure that all unit tests pass (for small modifications, it might be sufficient to only run the specific test file, e.g., `python -m py.test tests/kernels/test_graph_kernels.py`)
- [ ] Checked for style issues by running `black .` and `isort .`

### CODE REVIEW

File: gauche/gp.py [hunk 1]
```diff
@@ -52,9 +96,11 @@ def __init__(self, train_inputs, train_targets, likelihood):
...
         if train_inputs is not None:
             self.train_inputs = tuple(
-                i.unsqueeze(-1)
-                if torch.is_tensor(i) and i.ndimension() == 1
-                else i
+                (
+                    i.unsqueeze(-1)
+                    if torch.is_tensor(i) and i.ndimension() == 1
+                    else i
+                )
```
Comment (kkovary) [inline gauche/gp.py:103]: changes made by `make format`

File: gauche/gp.py [hunk 2]
```diff
@@ -71,9 +117,11 @@ def __call__(self, *args, **kwargs):
...
 
         inputs = [
-            i.unsqueeze(-1)
-            if torch.is_tensor(i) and i.ndimension() == 1
-            else i
+            (
+                i.unsqueeze(-1)
+                if torch.is_tensor(i) and i.ndimension() == 1
+                else i
+            )
```
Comment (kkovary) [inline gauche/gp.py:124]: changes made by `make format`

File: gauche/gp.py [hunk 3]
```diff
@@ -1,12 +1,56 @@
...
+
+Provides functionality for handling non-tensorial inputs, saving/loading models,
+and working with graph kernels in Gaussian Process models.
+
+Extends GPyTorch's ExactGP for structured, non-tensor inputs like graphs.
+"""
+
+import importlib
 from copy import copy, deepcopy
-from functools import lru_cache
```
Comment (kkovary) [inline gauche/gp.py:2]: unused import

**Review by Ryan-Rhys — APPROVED** [review]
LGTM! Happy to merge. Tagging @InfProbSciX for any last minute objections!

**Comment by kkovary** [comment]
FYI I'm going to be training some models using this framework this week and plan to try adding some additional atom level features. Based on how that goes, there may be a future PR related to this.

**Pull request merged**